traffic_ops/app/lib/API/DeliveryService/SslKeys.pm

```diff
@@ -120,12 +120,23 @@ sub view_by_xml_id {
 		if ( !$version ) {
 			$version = 'latest';
 		}
-		$key = "$key-$version";
+		my $ds = $self->db->resultset('Deliveryservice')->search( { xml_id => $xml_id })->single();
+		if (!$ds) {
+			return $self->alert( { Error => " - Could not found delivery service with xml_id=$xml_id!" } );
+		}
+		my $ds_id = $ds->id;
+		my $key = "ds_$ds_id-$version";
```

Comment:
As discussed in Slack, xml_id is immutable now, will revert this change in TC-547.